MDANSE/Src/MDANSE/Chemistry/ChemicalSystem.py

```diff
@@ -145,31 +163,28 @@ def get_substructure_matches(
         set[int]
             An set of matched atom indices.
         """
-        substruct_set = set()
         matches = self.rdkit_mol.GetSubstructMatches(
             Chem.MolFromSmarts(smarts), maxMatches=maxmatches
         )
-        for match in matches:
-            substruct_set.update(match)
-        return substruct_set
+        return {ind for match in matches for ind in match}
 
     @property
     def atom_list(self) -> list[str]:
-        """List of all non-ghost atoms in the ChemicalSystem."""
+        """Return the indices of all atoms in the ChemicalSystem."""
         return self._atom_types
 
     @property
     def name_list(self) -> list[str]:
-        """List of all non-ghost atoms in the ChemicalSystem."""
+        """Return the names of all atoms in the ChemicalSystem."""
         if self._atom_names is not None:
             return self._atom_names
         return self._atom_types
 
-    def atom_property(self, property: str) -> list[Any]:
-        """List of a specific property, for all atoms in the system"""
+    def atom_property(self, atom_property: str) -> list[Any]:
+        """Return the values of a specific property, for all atoms in the system."""
         lookup = {}
         for atom in self.atom_list:
-            lookup[atom] = self._database.get_atom_property(atom, property)
+            lookup[atom] = self._database.get_atom_property(atom, atom_property)
         return [lookup[atom] for atom in self.atom_list]
```

Comment:
Is this just ultimately:
```suggestion
        return [self._database.get_atom_property(atom, property) for atom in self.atom_list]
```